include/aos/common/pkcs11/pkcs11.hpp

```diff
@@ -0,0 +1,718 @@
+     * @param allocator allocator for token/session objects.
+     */
+    Utils(pkcs11::SessionContext& session, Allocator& allocator)
+    {
+        (void)session;
+        (void)allocator;
+    }
+
+    /**
+     * Creates an RSA key pair on the token.
+     *
+     * @param id token id.
+     * @param label label.
+     * @param bitsCount key size.
+     * @return RetWithError<SharedPtr<PrivateKey>>.
+     */
+    RetWithError<SharedPtr<PrivateKey>> GenerateRSAKeyPairWithLabel(
+        const Array<uint8_t>& id, const String& label, size_t bitsCount)
+    {
+        (void)id;
+        (void)label;
+        (void)bitsCount;
+
+        return {nullptr, Error::Enum::eNone};
+    }
+
+    /**
+     * Creates a ECDSA key pair on the token using curve.
+     *
+     * @param id token id.
+     * @param label label.
+     * @param curve elliptic curve.
+     * @return RetWithError<SharedPtr<PrivateKey>>.
+     */
+    RetWithError<SharedPtr<PrivateKey>> GenerateECDSAKeyPairWithLabel(
+        const Array<uint8_t>& id, const String& label, EllipticCurve curve)
+    {
+        (void)id;
+        (void)label;
+        (void)curve;
+
+        return {nullptr, Error::Enum::eNone};
+    }
+
+    /**
+     * Imports a certificate onto the token.
+     *
+     * @param id certificate id.
+     * @param label certificate label.
+     * @param cert certificate.
+     * @return Error.
+     */
+    Error ImportCertificate(const Array<uint8_t>& id, const String& label, const crypto::x509::Certificate& cert)
+    {
+        (void)id;
+        (void)label;
+        (void)cert;
+
+        return {};
+    }
+
+    /**
+     * Destroys a previously imported certificate.
+     *
+     * @param id certificate id.
+     * @param label certificate label.
+     * @return Error.
+     */
+    Error DeleteCertificate(const Array<uint8_t>& id, const String& label)
+    {
+        (void)id;
+        (void)label;
+
+        return {};
+    }
+
+    /**
+     * Checks whether certificate with specified attributes has been imported.
+     *
+     * @param issuer certificate issuer.
+     * @param serialNumber certificate serialNumber.
+     * @return Error.
+     */
+    RetWithError<bool> HasCertificate(const Array<uint8_t>& issuer, const Array<uint8_t>& serialNumber)
+    {
+        (void)issuer;
+        (void)serialNumber;
+
+        return {true};
+    }
+
+    /**
+     * Retrieves a previously created asymmetric key pair.
+     *
+     * @param id key id.
+     * @param label key label.
+     * @return RetWithError<SharedPtr<PrivateKey>>.
+     */
+    RetWithError<SharedPtr<PrivateKey>> FindPrivateKey(const Array<uint8_t>& id, const String& label)
+    {
+        (void)id;
+        (void)label;
+
+        return {nullptr};
+    }
+
+    /**
+     * Converts byte array to string.
+     *
+     * @param src source byte array.
+     * @param dst destination string.
+     * @return Error.
+     */
+    static Error Convert(const Array<uint8_t>& src, const String& dst)
```

Comment:
Don't understand why do you need this method.

Reply:
In order to convert Label to and from pkcs11 UTF8 string(which is a byter array). It is needed because certmodule pkcs11 adds custom SearchObject with label & uses low level FindObject/GetObjAttrib with it.

Reply:
Is it just cast aks uint8_t[] to char[]? Or uint8_t[] to string and back?

If it is first case then it should be handled in String and Array classes. We should have straight simple conversion between Array<uint8_t> and String. It would be useful in other cases as well.

For the second case it is already done on String class.